### ISSUE TITLE
certbot webroot confused by mod_rewrite host handling for port 80

### ISSUE THREAD

**Issue by daveola** [certbot/certbot#5929]
I run apache with a large number of domains.  I use mod_rewrite to handle hosts for port 80, but use specific confs for the few domains that I need SSL for on port 443.  Unfortunately certbot can't figure out my domain situation even if I use webroot.  It can get me the cert, but I need to install it manually.

## My operating system is (include version):
Ubuntu 16.04.4 LTS

## I installed Certbot with (certbot-auto, OS package manager, pip, etc):
certbot script

## I ran this command and it produced this output:
```
% certbot -i apache -a webroot -w /WWW/web/DavePics.com/ -d davepics.com
Saving debug log to /var/log/letsencrypt/letsencrypt.log
Plugins selected: Authenticator webroot, Installer apache
Obtaining a new certificate
Performing the following challenges:
http-01 challenge for davepics.com
Using the webroot path /WWW/web/DavePics.com for all unmatched domains.
Waiting for verification...
Cleaning up challenges
No vhost exists with servername or alias of davepics.com. No vhost was selected. Please specify ServerName or ServerAlias in the Apache config.
No vhost selected

IMPORTANT NOTES:
 - Unable to install the certificate
```

## Certbot's behavior differed from what I expected because:
It can't install the certificate.

###### Logs are stored in `/var/log/letsencrypt` by default. Feel free to redact domains, e-mail and IP addresses as you see fit.
```
2018-05-06 02:13:33,542:ERROR:certbot_apache.configurator:No vhost exists with servername or alias of davepics.com. No vhost was selected. Please specify ServerName or ServerAlias in the Apache config.
2018-05-06 02:13:33,558:DEBUG:certbot.error_handler:Encountered exception:
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/certbot/client.py", line 442, in deploy_certificate
    fullchain_path=fullchain_path)
  File "/usr/lib/python3/dist-packages/certbot_apache/configurator.py", line 302, in deploy_cert
    vhosts = self.choose_vhosts(domain)
  File "/usr/lib/python3/dist-packages/certbot_apache/configurator.py", line 326, in choose_vhosts
    return [self.choose_vhost(domain)]
  File "/usr/lib/python3/dist-packages/certbot_apache/configurator.py", line 507, in choose_vhost
    return self._choose_vhost_from_list(target_name, temp)
  File "/usr/lib/python3/dist-packages/certbot_apache/configurator.py", line 518, in _choose_vhost_from_list
    raise errors.PluginError("No vhost selected")
certbot.errors.PluginError: No vhost selected
```

## Here is the relevant nginx server block or Apache virtualhost for the domain I am configuring:
I even tried to temporarily setup a virtualhost for the given domain and restarted apache and still got the same error message, even though the host shows up in status:
```
% apache2ctl -S | grep davepics.com
65.111.161.40:80       davepics.com (/WWW/conf/sites-enabled/davepics-ssl.conf:3)
         port 443 namevhost davepics.com (/WWW/conf/sites-enabled/davepics-ssl.conf:2)
                 alias www.davepics.com
```

**Comment by schoen** [comment]
Hi @daveola, what version of Certbot are you running? And could you post the entire `davepics-ssl.conf` file?

**Comment by daveola** [comment]
I don't see any way to get version info from certbot.

This is ubuntu, so according to apt, the package version is 0.22.2-1+ubuntu16.04.1+certbot+1

The conf without comments is:
```
<IfModule mod_ssl.c>
        <VirtualHost *:443>
                ServerName davepics.com
                ServerAlias www.davepics.com
                DocumentRoot /WWW/web/DavePics.com
                SSLEngine on
                SSLCertificateFile      /WWW/conf/ssl/certs/davepics.cert
                SSLCertificateKeyFile /WWW/conf/ssl/private/davepics.private.key
                SSLCertificateChainFile /WWW/conf/ssl/certs/sub.class1.server.ca.pem

                <FilesMatch "\.(cgi|shtml|phtml|php)$">
                                SSLOptions +StdEnvVars
                </FilesMatch>
                <Directory /usr/lib/cgi-bin>
                                SSLOptions +StdEnvVars
                </Directory>

                BrowserMatch "MSIE [2-6]" \
                                nokeepalive ssl-unclean-shutdown \
                                downgrade-1.0 force-response-1.0
                BrowserMatch "MSIE [17-9]" ssl-unclean-shutdown

        </VirtualHost>
</IfModule>
```

**Comment by joohoi** [comment]
Hi! Is this issue still relevant? We've made some changes to `VirtualHost` handling during the year, it could well be that this has been fixed since version 0.22.

**Comment by daveola** [comment]
I found some sort of workaround, and I don't actually remember what it is, but things are working again, so I'll just close this.
